### PR TITLE
R4R: implement Kava client

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR implements KavaClient, a client that enables interaction with Kava via a rest server. It includes the core structure of the client and a README demonstrating usage. This PR is R4R.

TODOs include:
- ~~test client-side generation of swap IDs~~ - https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/60552dddc29fc56c16038c6512286f3e14ccfd83
- ~~fix bug with createSwap. error message: "signature verification failed; verify correct account sequence and chain-id: unauthorized"~~ - https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/60552dddc29fc56c16038c6512286f3e14ccfd83
- ~~standardize usage of `utils.loadCoins(asset, amount)` for correct decimal precision~~ -https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/60552dddc29fc56c16038c6512286f3e14ccfd83

### CODE REVIEW

File: src/client/index.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+      throw new Error("mnemonic cannot be undefined");
+    }
+    this.wallet = sig.createWalletFromMnemonic(
+      mnemonic,
+      "",
+      prefix,
+      derivationPath
+    );
+    return this;
+  }
```
Comment (rhuairahrighairidh) [inline src/client/index.js:86]: If the other wallet args are passed as optional args with defaults then you can still just call `setWallet("some mnemonic")` in most cases, but more advanced users can pass in the other bits as needed.

```suggestion
  setWallet(mnemonic, derivationPath=derivationPath , password="", prefix=prefix) {
    if (!mnemonic) {
      throw new Error("mnemonic cannot be undefined");
    }
    this.wallet = sig.createWalletFromMnemonic(
      mnemonic,
      "",
      prefix,
      derivationPath
    );
    return this;
  }
```

File: src/tx/index.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const _ = require("lodash");
+const axios = require("axios");
+
+const api = {
+  getAccount: "/auth/accounts",
+  postTx: "/txs"
+};
+
+async function getTx(path, base) {
+  return await axios.get(new URL(path, base).toString());
```
Comment (karzak) [inline src/tx/index.js:11]: on node `v8.10.0` I get an error here saying URL is not defined. Adding
```
const URL = require('url').URL
```
to imports fixed it

File: src/client/index.js (hunk 1 above)
Comment (denalimarsh) [reply src/client/index.js:86]: Should the full derivation path be parameterized or just the coin ID?

Comment (denalimarsh) [reply src/client/index.js:86]: What do you think about adding ledger support? Not within the scope of this PR, but making an issue for it and adding in the future.

Comment (denalimarsh) [reply src/client/index.js]: https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/787d9274b3095e334872e9f012e3856595323cf4

File: src/tx/index.js (hunk 2 above)
Comment (denalimarsh) [reply src/tx/index.js:12]: https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/3983081920e166eea012ca5ee1a0700598ee2bf2

**Comment by denalimarsh** [comment]
> LGTM. +1 to adding issue for ledger support.

Created an issue for ledger support https://github.com/Kava-Labs/javascript-sdk/issues/4

**Pull request merged**